src/routes/unregisterPublicKeys.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  AuthorizedRequest,
+  Env,
+  ProfileWithId,
+  UnregisterPublicKeyRequest,
+  UnregisterPublicKeyResponse,
+} from '../types'
+import {
+  KnownError,
+  getProfileFromPublicKey,
+  getProfilePublicKeys,
+  incrementProfileNonce,
+  removeProfilePublicKeys,
+} from '../utils'
+
+export const unregisterPublicKeys = async (
+  request: AuthorizedRequest<UnregisterPublicKeyRequest>,
+  env: Env
+) => {
+  const respond = (status: number, response: UnregisterPublicKeyResponse) =>
+    new Response(JSON.stringify(response), {
+      status,
+    })
+
+  const {
+    auth: { publicKey },
+    publicKeys,
+  } = request.parsedBody.data
+
+  // Get existing profile.
+  let profile: ProfileWithId | undefined
+  try {
+    profile = await getProfileFromPublicKey(env, publicKey)
+  } catch (err) {
+    console.error('Profile retrieval', err)
+
+    return respond(500, {
+      error:
+        'Failed to retrieve existing profile: ' +
+        (err instanceof Error ? err.message : `${err}`),
+    })
+  }
+
+  if (!profile) {
+    return respond(404, {
+      error: 'Profile not found.',
+    })
+  }
+
+  // Validate that all public keys are attached to this profile.
+  const profilePublicKeys = (await getProfilePublicKeys(env, profile.id)).map(
+    (row) => row.publicKey
+  )
+  if (publicKeys.some((key) => !profilePublicKeys.includes(key))) {
+    return respond(401, {
+      error: 'Not all public keys are attached to this profile.',
```

Comment:
When might this happen?

Reply:
it should never happen unless someone is misusing the API. it's just a validation to ensure it's used correctly.